Traceback (most recent call last):
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/micromamba-root/envs/comp2421-book/lib/python3.12/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def f(x):
    return x * x - 2


a = 0
b = 2
tol = 1.0e-4

fa = f(a)
fb = f(b)

it = 0

headers = ["it", "a", "b", "f(a)", "f(b)", "update"]
data = []

while b - a > tol:
    new_data = [it, a, b, f(a), f(b)]

    c = (a + b) / 2
    fc = f(c)

    if fa * fc < 0:
        b = c
        fb = fc

        new_data.append(f"b := {c:.4f}")
    else:
        a = c
        fa = fc

        new_data.append(f"a := {c:.4f}")

    data.append(new_data)
    it += 1

import pandas as pd

df = pd.DataFrame(data, columns=headers)
df.style.hide_index().set_caption("Results of bisection method on example 1")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[3], line 40[0m
[1;32m     37[0m [38;5;28;01mimport[39;00m [38;5;21;01mpandas[39;00m [38;5;28;01mas[39;00m [38;5;21;01mpd[39;00m
[1;32m     39[0m df [38;5;241m=[39m pd[38;5;241m.[39mDataFrame(data, columns[38;5;241m=[39mheaders)
[0;32m---> 40[0m [43mdf[49m[38;5;241;43m.[39;49m[43mstyle[49m[38;5;241;43m.[39;49m[43mhide_index[49m()[38;5;241m.[39mset_caption([38;5;124m"[39m[38;5;124mResults of bisection method on example 1[39m[38;5;124m"[39m)

[0;31mAttributeError[0m: 'Styler' object has no attribute 'hide_index'

